src/lists.py

```diff
@@ -69,8 +90,14 @@ def take(x: LList[T], k: int) -> LList[T]:
     >>> take(Link(1, Link(2, Link(3, None))), 2)
     Link(1, Link(2, None))
     """
-    ...
-
+    if x is None:
+        return None
+    y = None
+    for index, item in enumerate(x):
+        if index >= k:
+            return y
+        y = reverse(Link(item, y))
```

Comment:
You are reversing `y` every time you do this. Well, the previous `y`. If you do it an even number of times, I suppose it could be fine (except that the running time is quadratic instead of linear), but I could easily see this failing.

Why not wait with reversing until you return?

Reply:
Thanks for the feedback!